### PR TITLE
[IMPROVEMENT] Use update_status_on_github to report status 

### PR DESCRIPTION
**Pull request by Tarun-Arora**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [x] I am an active contributor to the project.

---

This PR is to strictly enforce use of `update_status_on_github` to update status on github
And fix blocked_users endpoint and increase test coverage

### CODE REVIEW

**Comment by Tarun-Arora** [comment]
Issue #825 is causing the GH Actions to fail

File: mod_ci/controllers.py [hunk 1]
```diff
@@ -935,14 +935,12 @@ def start_ci():
                     g.db.add(progress)
                     g.db.commit()
                     # If test run status exists, mark them as cancelled
+                    gh_commit = repository.get_commit(test.commit)
                     for status in repository.get_commit(test.commit).get_statuses():
```
Comment (thealphadollar) [inline mod_ci/controllers.py:939]: The new line "gh_commit=..." should come before the comment line "# If test...", right?

**Review by thealphadollar — DISMISSED** [review]
LGTM. There's a small query I've left, please have a look.

File: mod_ci/controllers.py (hunk 1 above)
Comment (Tarun-Arora) [reply mod_ci/controllers.py:939]: Oh yes, thanks.
I would change this.

**Review by thealphadollar — APPROVED** [review]

**Pull request merged**